lib/singlepass-backend/src/codegen_x64.rs

```diff
@@ -880,6 +893,92 @@ impl ModuleCodeGenerator<X64FunctionCode, X64ExecutionContext, CodegenError>
         a.emit_label(label);
         labels.insert(id, (label, Some(offset)));
 
+        // Singlepass internally treats all arguments as integers, but the standard System V calling convention requires
+        // floating point arguments to be passed in XMM registers.
+        //
+        // FIXME: This is only a workaround. We should fix singlepass to use the standard CC.
+        let sig = self
+            .signatures
+            .as_ref()
+            .expect("signatures itself")
+            .get(sigindex)
+            .expect("signatures");
+        // Translation is expensive, so only do it if needed.
+        if sig
+            .params()
+            .iter()
+            .find(|&&x| x == Type::F32 || x == Type::F64)
+            .is_some()
+        {
+            let mut param_locations: Vec<Location> = vec![];
+
+            // Allocate stack space for arguments.
+            let stack_offset: i32 = if sig.params().len() > 5 {
+                5 * 8
+            } else {
+                (sig.params().len() as i32) * 8
+            };
+            if stack_offset > 0 {
+                a.emit_sub(
```

Comment:
Note that x86-64 SysV ABI requires a 16-byte stack alignment, not 8-byte. While a fault due to misaligment is rare, it could happen if the callee uses aligned movs to access the stack. (I'm not sure if this is a bug because I don't know whether something else is guaranteed to realign the stack between here and the point where we next enter non-singlepass-generated code. Doesn't have to be an import, could also be a vmcalls function.)

Reply:
The call to the actual import is a tail call and the stack space allocated here will be released before jumping to the target function.